wagtail_transfer/richtext.py

```diff
@@ -0,0 +1,85 @@
+import re
+
+from wagtail.core.rich_text.feature_registry import FeatureRegistry
+from wagtail.core.rich_text.rewriters import extract_attrs
+
+from .models import get_base_model
+
+features = FeatureRegistry()
+
+FIND_A_TAG = re.compile(r'<a(\b[^>]*)>')
+FIND_EMBED_TAG = re.compile(r'<embed(\b[^>]*)/>')
+FIND_ID = re.compile(r'id="([^"]*)"')
+
+
+class RichTextReferenceHandler:
+    """
+    Handles updating ids and retrieving object references for <tag type_attribute="foo" id="my_id" /> tags representing references within rich text. Tags are found using a regex tag_matcher, and their
+    specific handler (eg PageLinkHandler) is looked up using a dict, handlers, which maps the value of type_attribute (where type_attribute might be emebedtype)
+    to a specific handler.
+    """
+    def __init__(self, handlers, tag_matcher, type_attribute, destination_ids_by_source={}):
+        self.handlers = handlers
+        self.tag_matcher = tag_matcher
+        self.type_attribute = type_attribute
+        self.destination_ids_by_source = destination_ids_by_source
```

Comment:
Passing `destination_ids_by_source` as a parameter to `update_ids` instead of making it part of RichTextReferenceHandler's state would simplify this code quite a lot, and stop me from feeling antsy about race conditions (what happens if two import processes try to work with the global reference_handler object at the same time) :-)

Reply:
Yes, I didn't think about two import processes - I was assuming context would be the same. I've changed that now.